ui/src/shared/utils/resourceToTemplate.ts

```diff
@@ -96,8 +96,87 @@ export const taskToTemplate = (
   return template
 }
 
-// Todo: update template to match task format
-export const dashboardToTemplate = (dashboard: Dashboard) => ({
-  meta: {chronografVersion: '2.0'},
-  dashboard,
+const viewToIncluded = (view: View) => ({
+  type: TemplateType.View,
+  id: view.id,
+  attributes: view
+})
+const viewToRelationship = (view: View) => ({
+  type: TemplateType.View, id: view.id
 })
+
+
+const cellToIncluded = (cell: Cell, views: View[]) =>  {
+  const cellView = views.find(v => v.id === cell.id)
+  const viewRelationship = viewToRelationship(cellView)
+
+  return {
+    id: cell.id,
+    type: TemplateType.Cell,
+    attributes: cell,
+    relationships: {
+      [TemplateType.View]: {
+        data: viewRelationship
+      }
+    }
+  }
+}
+
+const cellToRelationship = (cell: Cell) => ({
+  type: TemplateType.Cell, id: cell.id
+})
+
+// Todo: update template to match task format
+export const dashboardToTemplate = (
+  dashboard: Dashboard,
+  views: View[],
+  baseTemplate = blankTaskTemplate(),
+) => {
+  const dashboardName = _.get(dashboard, 'name', '')
+  const templateName = `${dashboardName}-Template`
+
+  const dashboardAttributes = _.pick(dashboard, [
+    'name',
+    'description',
+  ])
+
+  const labels = getDeep<Label[]>(dashboard, 'labels', [])
+  const includedLabels = labels.map(l => labelToIncluded(l))
+  const relationshipsLabels = labels.map(l => labelToRelationship(l))
+
+  const cells = getDeep<Cell[]>(dashboard, 'cells', [])
+  const includedCells = cells.map(c => cellToIncluded(c, views))
+  const relationshipsCells = cells.map(c => cellToRelationship(c))
+
+  const includedViews = views.map(v => viewToIncluded(v))
+
+  const template = {
+    ...baseTemplate,
+    meta: {
+      ...baseTemplate.meta,
+      name: templateName,
+      version: CURRENT_TEMPLATE_VERSION,
+      description: `template created from dashboard: ${dashboardName}`,
+    },
+    content: {
+      ...baseTemplate.content,
+      data: {
+        ...baseTemplate.content.data,
+        type: TemplateType.Dashboard,
```

Comment:
so I had put type in the baseTemplate for a dashboard, but I'm not sure that it has much utility anymore. I would love to test this but we don't have tests in client.